codice/ddf
# DDF-4228 List Items query local first

#### Notes
https://github.com/codice/ddf/commit/eac285bcfb5b27cc6680985b28a2241caaa86ddb - Changing `Query.js` from define to using require statements. This also reduces al l indentations, meaning the file looks like it has completely changed. To make it easy to review, all logic prevalent to this ticket is in https://github.com/codice/ddf/commit/9e09a656e98267f6c080dffc52bad7071bc53da8.

Please only review https://github.com/codice/ddf/commit/9e09a656e98267f6c080dffc52bad7071bc53da8.

#### What does this PR do?
Lists should query local before attempting to make a federated query because the majority of list items are stored locally.

#### Who is reviewing it? 

@millerw8 
@bdeining 
@troymohl 

#### Select relevant component teams: 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@djblue

#### How should this be tested?

1. Upload to a list by manually creating a metacard using the builder.
2. Setup a remote source, search, and then add one of the items to the list.
3. Verify that there are 2 sets of network requests, one for local and one for remote sources.


#### Any background context you want to provide?


#### What are the relevant tickets?
[DDF-4228](https://codice.atlassian.net/browse/DDF-4228)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
